python-client/giskard/ml_worker/websocket/listener.py

```diff
@@ -0,0 +1,838 @@
+    for arg in request_arguments:
+        if arg.model:
+            filled_info = template_info.replace("xxx", arg.name)
+            info["suffix"] += filled_info.replace(arg.name + "_id", arg.model.id)
+            info["project_key"] = arg.model.project_key  # in case model is in the args and dataset is not
+        elif arg.dataset:
+            filled_info = template_info.replace("xxx", arg.name)
+            info["suffix"] += filled_info.replace(arg.name + "_id", arg.dataset.id)
+            info["project_key"] = arg.dataset.project_key  # in case dataset is in the args and model is not
+    return info
+
+
+def map_function_meta_ws(callable_type):
+    return {
+        test.uuid: websocket.FunctionMeta(
+            uuid=test.uuid,
+            name=test.name,
+            displayName=test.display_name,
+            module=test.module,
+            doc=test.doc,
+            code=test.code,
+            moduleDoc=test.module_doc,
+            tags=test.tags,
+            type=test.type,
+            args=[
+                websocket.TestFunctionArgument(
+                    name=a.name,
+                    type=a.type,
+                    optional=a.optional,
+                    default=str(a.default),
+                    argOrder=a.argOrder,
+                )
+                for a in test.args.values()
+            ],
+        )
+        for test in tests_registry.get_all().values()
+        if test.type == callable_type
+    }
+
+
+def log_artifact_local(local_file, artifact_path=None):
+    # Log artifact locally from an internal worker
+    verify_artifact_path(artifact_path)
+
+    file_name = os.path.basename(local_file)
+
+    paths = (projects_dir, artifact_path, file_name) if artifact_path else (projects_dir, file_name)
+    artifact_file = posixpath.join("/", *paths)
+    Path(artifact_file).parent.mkdir(parents=True, exist_ok=True)
+
+    shutil.copy(local_file, artifact_file)
+
+
+def map_dataset_process_function_meta_ws(callable_type):
+    return {
+        test.uuid: websocket.DatasetProcessFunctionMeta(
+            uuid=test.uuid,
+            name=test.name,
+            displayName=test.display_name,
+            module=test.module,
+            doc=test.doc,
+            code=test.code,
+            moduleDoc=test.module_doc,
+            tags=test.tags,
+            type=test.type,
+            args=[
+                websocket.TestFunctionArgument(
+                    name=a.name,
+                    type=a.type,
+                    optional=a.optional,
+                    default=str(a.default),
+                    argOrder=a.argOrder,
+                )
+                for a in test.args.values()
+            ],
+            cellLevel=test.cell_level,
+            columnType=test.column_type,
+            processType=test.process_type.name,
+        )
+        for test in tests_registry.get_all().values()
+        if test.type == callable_type
+    }
+
+
+@websocket_actor(MLWorkerAction.getInfo)
```

Comment:
this file is a bit huge, I'd split it by low level utilities (actiond_in_tread, etc.) and the actors themselves. Plus you can group actors into separate submodules to keep it more organized

Reply:
Done, but actors can be organised in further steps.